### PR TITLE
split different ide feature

### PR DESCRIPTION
**Pull request by wsdjeg**
maybe it is better do not use .classpath with gradle together.
should we provide an option,let the user choose by themselves？
for example!
```vim
if !existe('g:javacomplete_ide_feature')||g:javacomplet_ide_feature == 'eclispe'
     return classpath generated from .classpath
else if !existe('g:javacomplete_ide_feature')||g:javacomplet_ide_feature == 'maven'
     return classpath generated from pom.xml
else if !existe('g:javacomplete_ide_feature')||g:javacomplet_ide_feature == 'gradle'
    return classpath generated from build.gradle
endif
```

### CODE REVIEW

**Comment by wsdjeg** [comment]
this PR also fix import android api in gradle project



## before this PR

**Comment by wsdjeg** [comment]
@DonnieWest now the plugin works well with the gradle project,i think i can add completion for R class,please waiting

**Comment by wsdjeg** [comment]
greate it works very well
@DonnieWest

**Comment by wsdjeg** [comment]
@artur-shaik  this can be merged safety

**Comment by wsdjeg** [comment]
has been tested with this project
https://github.com/wsdjeg/cube-sdk

**Comment by wsdjeg** [comment]
please watting !
after last commit can not import android API

**Comment by wsdjeg** [comment]
the issue of can not import android API is not due to this PR, before or after this PR,import only works well at the first time running maybe it is due to there are too many class

**Comment by wsdjeg** [comment]
i have checked the PR by use the same func for syntastic,here is my fork 
https://github.com/wsdjeg/syntastic
and in gradle project ,it can auto compile java file same as maven without error,so I think it can be merged safety

**Comment by richox** [comment]
i think it's better to separate the classpath generator into a independent plugin, so that other java plugins can use it too.

**Comment by wsdjeg** [comment]
i know,and i am tring to do,before it ,let 's make jc2 works better,also syc do not want to merge my PR
https://github.com/scrooloose/syntastic/pull/161

**Comment by wsdjeg** [comment]
also maybe there is no need create a new plugin ,just let JC2 provide some info api of current project
https://github.com/artur-shaik/vim-javacomplete2/issues/129

**Comment by wsdjeg** [comment]
the import issue really due to there are too many classes, when i open the file `~/.javacomplete2/classpath/_home_wsdjeg_fork_cube-sdk_build_gradle` and delete some jars and  the import works well,also if i do not add the gradle build out put dir into class path ,the complete after `import android._`also works well

**Comment by wsdjeg** [comment]
i think i have find the error!

**Comment by wsdjeg** [comment]
hi @artur-shaik i think must be some error in javavi,because ,if i add this into classpath `/home/wsdjeg/fork/cube-sdk/core/build/intermediates/classes/debug`
all the import does not work except the package name which is begin with `in`
you see

**Comment by wsdjeg** [comment]
@artur-shaik if I add the compile outdir of gradle into classpath ,the import will be break ,do you know why , only can import same package name as the current project.

**Comment by artur-shaik** [comment]
There was one bug, did you try it with last commits?

**Comment by wsdjeg** [comment]
i have tried ,still can not import

**Comment by wsdjeg** [comment]
strange.after `import android` i type dot show me `Pattern not found` then i type `backspace` type dot again still `Pattern not found` but if i type third times dot i will get the results. then every thing will be ok

**Comment by artur-shaik** [comment]
Hm, seems like issue with `*.class` files search. I think it should be optimized somehow.

**Comment by wsdjeg** [comment]
Do you use neocomplete,i find it support java better,because YCM just support its core complete better.

**Comment by artur-shaik** [comment]
I use YCM, but this issue related to javacomplete.

**Comment by wsdjeg** [comment]
i know ,i just finish readding neocomplete's doc.it is too complex

File: autoload/javacomplete.vim [hunk 1]
```diff
@@ -232,6 +233,12 @@ function! s:GenerateGradleClassPath(path, gradle) abort
     let ret = system(gradle . ' -q -I ' . shellescape(f) . ' classpath' )
     if v:shell_error == 0
       let cp = filter(split(ret, "\n"), 'v:val =~ "^CLASSPATH:"')[0][10:]
+      if filereadable(getcwd() . "/build.gradle")
```
Comment (wsdjeg) [inline autoload/javacomplete.vim]: @DonnieWest  maybe this six lines should be deleted

**Pull request closed without merging**